synapse/lib/stormhttp.py

```diff
@@ -240,12 +258,24 @@ async def inetHttpConnect(self, url, headers=None, ssl_verify=True, timeout=300)
             await sock.fini()
             return s_common.retnexc(e)
 
-    async def _httpRequest(self, meth, url, headers=None, json=None, body=None, ssl_verify=True,
-                           params=None, timeout=300, allow_redirects=True, ):
+    def _buildFormData(self, fields):
+        data = aiohttp.FormData()
+        for field in fields:
+            data.add_field(field.get('name'),
+                           field.get('value'),
+                           content_type=field.get('content_type'),
+                           filename=field.get('filename'),
+                           content_transfer_encoding=field.get('content_transfer_encoding'))
+        return data
+
+    async def _httpRequest(self, meth, url, headers=None, json=None, body=None,
+                           fields=None, ssl_verify=True, params=None, timeout=300,
```

Comment:
Fields need to be added to the end of the arguments list, otherwise we risk breaking users who are calling this without keyword arguments.

Reply:
fixed